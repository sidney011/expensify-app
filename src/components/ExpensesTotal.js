import React from 'react';
import { connect } from 'react-redux';
import selectExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses'
import numeral from 'numeral';

export const ExpensesTotal = ({expensesCount, expensesTotal}) => (
    <div>
        <h2>Viewing {expensesCount} {expensesCount > 1 ? 'expenses' : 'expense'} totalling {numeral(expensesTotal / 100).format('$0,0.00')}</h2>
    </div>
);

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);

    return {
        expensesTotal: selectExpensesTotal(state.expenses),
        expensesCount: visibleExpenses.length
    };
};

export default connect(mapStateToProps)(ExpensesTotal);

