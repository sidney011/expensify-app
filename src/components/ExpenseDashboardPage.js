import React from "react";
import ExpenseList from './ExpenseList';
import ExpensesTotal from './ExpensesTotal';
import ExpenseListFilters from './ExpenseListFilters';


const ExpenseDashboardPage = () => (
    <div>
        <ExpensesTotal />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;