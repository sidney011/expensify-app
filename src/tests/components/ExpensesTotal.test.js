import React from 'react';
import { shallow } from 'enzyme';
import {Â ExpensesTotal } from '../../components/ExpensesTotal';

test('should correctly render ExpensesTotal with 1 expense', () => {
    const wrapper = shallow(<ExpensesTotal expensesCount={1} expensesTotal={235}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesTotal with multiple expenses', () => {
    const wrapper = shallow(<ExpensesTotal expensesCount={23} expensesTotal={23393}/>);
    expect(wrapper).toMatchSnapshot();
});