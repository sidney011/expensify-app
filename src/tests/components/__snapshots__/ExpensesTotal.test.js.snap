// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesTotal with 1 expense 1`] = `
<div>
  <h2>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h2>
</div>
`;

exports[`should correctly render ExpensesTotal with multiple expenses 1`] = `
<div>
  <h2>
    Viewing 
    23
     
    expenses
     totalling 
    $233.93
  </h2>
</div>
`;
